cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,119 +11,149 @@
     private final List<CommandInvocationNode> nodes;
     private int position;
     private final Map<String, Set<CommandInvocationNode>> variables;
-    private final Map<String, LogicalBlock> subroutines = new HashMap<>();
+    private final Set<CommandInvocationNode> suspiciousPoints;
+    private final Map<String, AbstractSimulator.CommandSimulator> dynamicSimulators = new HashMap<>();
 
-    SimulationState(List<CommandInvocationNode> nodes, int position) {
-        this(nodes, position, new HashMap<>());
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints) {
+        this(nodes, position, suspiciousPoints, new HashMap<>());
     }
 
-    SimulationState(List<CommandInvocationNode> nodes, int position, Map<String, Set<CommandInvocationNode>> variables) {
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints, Map<String, Set<CommandInvocationNode>> variables) {
         this.nodes = nodes;
         this.position = position;
         this.variables = variables;
+        this.suspiciousPoints = suspiciousPoints;
     }
 
-    void simulate(Set<Node> suspiciousPoints, Node node) {
-        node.visitAll(new NodeVisitorAdapter() {
-            @Override
-            public void accept(CommandInvocationNode node) {
-                processUsages(node);
+    String getValue(Node argumentNode) {
+        return getValue(argumentNode, Collections.emptyMap());
+    }
 
-                Function<CommandInvocationNode, Node> setter = Checker.setters.get(node.getCommandName());
-                if (setter != null) {
-                    Node argument = setter.apply(node);
+    /** todo think of better name */
+    String getValue(Node argumentNode, Map<String, Node> substitutions) {
+        StringBuilder sb = new StringBuilder();
 
-                    if (argument != null) {
-                        String variable = ((ArgumentNode) argument).getArgument();
-                        // todo if variable is a reference through other variables we should inline it.
-                        variables.put(variable, Collections.singleton(node));
-                    }
-                } else if (node.getCommandName().equals("unset")) {
-                    List<Node> arguments = node.getArguments();
-                    if (arguments.size() > 0) {
-                        String variable = ((ArgumentNode) arguments.get(0)).getArgument();
-                        // todo if variable is a reference through other variables we should inline it.
+        argumentNode.visitAll(new NodeVisitorAdapter() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 32 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)